perception/object_recognition/tracking/multi_object_tracker/CMakeLists.txt

```diff
@@ -38,20 +33,28 @@ add_executable(multi_object_tracker_node
   src/data_association/successive_shortest_path/successive_shortest_path.cpp
 )
 
-add_dependencies(multi_object_tracker_node
-  ${${PROJECT_NAME}_EXPORTED_TARGETS}
-  ${catkin_EXPORTED_TARGETS}
+set(MULTI_OBJECT_TRACKER_HEADERS
+  src/data_association/successive_shortest_path/successive_shortest_path.hpp
+  include/multi_object_tracker/data_association/data_association.hpp
+  include/multi_object_tracker/tracker/tracker.hpp
+  include/multi_object_tracker/tracker/model/tracker_base.hpp
+  include/multi_object_tracker/tracker/model/bicycle_tracker.hpp
+  include/multi_object_tracker/tracker/model/pedestrian_tracker.hpp
+  include/multi_object_tracker/tracker/model/vehicle_tracker.hpp
+  include/multi_object_tracker/utils/utils.hpp
+  include/multi_object_tracker/multi_object_tracker_core.hpp
 )
 
-target_link_libraries(multi_object_tracker_node ${catkin_LIBRARIES})
-
-install(TARGETS multi_object_tracker_node
-        ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-        LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-        RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION})
+ament_auto_add_library(multi_object_tracker_node SHARED 
+  ${MULTI_OBJECT_TRACKER_SRC} 
+  ${MULTI_OBJECT_TRACKER_HEADERS}
+)
 
-install(
-  DIRECTORY
-    launch
-  DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
+rclcpp_components_register_node(multi_object_tracker_node
+  PLUGIN "MultiObjectTracker"
+  EXECUTABLE multi_object_tracker
```

Comment:
We decided not to convert to rclcpp_components at initial porting.
We don't have to revert this PR, but when you port other packages, you don't have to make them into component nodes.

Reply:
Yeah, sorry this was the last PR with `rclcpp_components`. This package was ported a while ago when we were in limbo about the usage of components. Subsequent packages use the hand-written main file.